File: TP1/src/boucles.py
"""
boucles.py

Objectif :
Dessine un triangle en alternant (* et #) pour le remplissage avec une taille saisi par l'utilisateur

Auteur :
Mohamed AIT EL HADJ

Date :
16/09/2024
"""

def for_triangle() :
    compteur = int(input("Entrez la taille du triangle : "))

    for i in range(1, compteur + 1):
        for j in range(1, i + 1):
            if i == 1 or i == compteur or j == 1 or j == i:
                print("*", end=" ")
            else:
                print("#", end=" ")
        print()

def while_triangle() :
    compteur = int(input("Entrez la taille du triangle : "))
    i = 1

    while i <= compteur:
        j = 1
        while j <= i:
            if i == 1 or i == compteur or j == 1 or j == i:
                print("*", end=" ")
            else:
                print("#", end=" ")
            j += 1
        print()
        i += 1

while_triangle()

File: TP1/src/boucles.c
#include <stdio.h>
#include "boucles.h"

/*
* boucles.c
*
* Objectif :
* Dessine un triangle en alternant (* et #) pour le remplissage avec une taille saisi par l'utilisateur
*
* Auteur :
* Mohamed AIT EL HADJ
*
* Date :
* 16/09/2024
*/

void for_triangle() {
    int compteur, i, j;
    printf("Entrez la taille du triangle : ");
    scanf("%d", &compteur);

    for (i = 1; i <= compteur; i++) {
        for (j = 1; j <= i; j++) {
            if (i == 1 || i == compteur || j == 1 || j == i) {
                printf("* ");
            } else {
                printf("# ");
            }
        }
        printf("\n");
    }
}

void while_triangle() {
    int compteur, i = 1, j;
    printf("Entrez la taille du triangle : ");
    scanf("%d", &compteur);

    while (i <= compteur) {
        j = 1;
        while (j <= i) {
            if (i == 1 || i == compteur || j == 1 || j == i) {
                printf("* ");
            } else {
                printf("# ");
            }
            j++;
        }
        printf("\n");
        i++;
    }
}

